### PR TITLE
Adds support for decimal qualifiers. Resolves #91

### LINKED ISSUE #91: [Bug] Support for decimal qualifiers 'M' & 'm'
I run into a problem when I tried to evaluate an expression with a ```decimal``` value (eg ```"0.5m"```).
```ExpressionParser``` detects it as ```double```, then ```PromoteExpression``` method try to convert it to ```decimal``` but keeps qualifier 'm' as part of text, and ```decimal.TryParse("0.11m", out e)``` returns 0.

### PR DESCRIPTION
**Pull request by pferraris**
I also added a test for real numbers and checked that all tests pass successfully.

### CODE REVIEW

File: test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs [hunk 1]
```diff
@@ -193,6 +193,23 @@ public void ParseLambda_ParameterName_Null()
         }
 
         [Fact]
+        public void ParseLambda_RealNumbers()
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/DynamicExpressionParserTests.cs]: Please order the test alphabetically in the file.

File: test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs [hunk 2]
```diff
@@ -657,6 +657,38 @@ public void ExpressionTests_FloatQualifiers_Negative()
         }
 
         [Fact]
+        public void ExpressionTests_DecimalQualifiers()
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs]: Please order the test alphabetically in the file.

File: test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs [hunk 3]
```diff
@@ -657,6 +657,38 @@ public void ExpressionTests_FloatQualifiers_Negative()
...
+            var result1 = values.Where("it == 2M or it == 3m");
+            var result2 = values.Where("it == 2.0M or it == 3.00m");
+
+            //Assert
+            Assert.Equal(resultValues.ToArray(), result1.ToArray());
+            Assert.Equal(resultValues.ToArray(), result2.ToArray());
+        }
+
+        [Fact]
+        public void ExpressionTests_DecimalQualifiers_Negative()
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/ExpressionTests.cs]: Please order the test alphabetically in the file.

File: test/System.Linq.Dynamic.Core.Tests/TextParserTests.cs [hunk 4]
```diff
@@ -132,6 +132,30 @@ public void TextParser_Parse_RealLiteralPlus()
         }
 
         [Fact]
+        public void TextParser_Parse_RealLiteralFloatQualifier()
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/TextParserTests.cs]: Please order the test alphabetically in the file.

File: test/System.Linq.Dynamic.Core.Tests/TextParserTests.cs [hunk 5]
```diff
@@ -132,6 +132,30 @@ public void TextParser_Parse_RealLiteralPlus()
...
+            var textParser = new TextParser(" 12.5f ");
+
+            // Assert
+            Check.That(textParser.CurrentToken.Id).Equals(TokenId.RealLiteral);
+            Check.That(textParser.CurrentToken.Pos).Equals(1);
+            Check.That(textParser.CurrentToken.Text).Equals("12.5f");
+        }
+
+        [Fact]
+        public void TextParser_Parse_RealLiteralDecimalQualifier()
```
Comment (StefH) [inline test/System.Linq.Dynamic.Core.Tests/TextParserTests.cs]: Please order the test alphabetically in the file.

**Comment by StefH** [comment]
Thanks.

Can you take a look at the review?

**Comment by pferraris** [comment]
It's done! I already ordered the tests, except for ParseLambda_IllegalMethodCall_ThrowsException which is below all the others.

**Pull request merged**